### PR TITLE
Flatten the schema search state and result into single classes used f…

### PR DESCRIPTION
**Pull request by abrooksv**
…or both all and single registry

* Simplifies a lot of code, removes generics
* Removed the wrapper SchemaSearchResultVersion

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/schemas/search/SchemaSearchDialog.kt [hunk 1]
```diff
@@ -4,48 +4,39 @@
...
     message("schemas.search.header.text.singleRegistry", registryName),
-    onCancelCallback,
-    alarmThreadToUse
+    onCancelCallback
 ) {
 
-    override fun getCurrentState(): SchemaSearchSingleRegistryDialogState {
-        val searchResults = IntStream.range(0, resultsModel.size()).mapToObj(resultsModel::get).toList()
-        return SchemaSearchSingleRegistryDialogState(currentSearchText(), searchResults, selectedSchema(), selectedSchemaVersion()?.version)
+    override fun createResultRenderer(): NotNullFunction<SchemaSearchResultWithRegistry, JComponent> = NotNullFunction {
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/schemas/search/SchemaSearchDialog.kt]: is the NotNullFunction required in the base?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/schemas/search/SchemaSearchDialog.kt [hunk 2]
```diff
@@ -55,32 +46,23 @@ class SchemaSearchSingleRegistryDialog(
 
 class SchemaSearchAllRegistriesDialog(
     project: Project,
-    onCancelCallback: (SchemaSearchAllRegistriesDialogState) -> Unit,
+    onCancelCallback: (SchemaSearchDialogState) -> Unit,
```
Comment (hunterwerlla) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/schemas/search/SchemaSearchDialog.kt]: nit: having the cancel callback in the middle is weird (even though it was already there), can we move it?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/schemas/search/SchemaSearchDialog.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/schemas/search/SchemaSearchDialog.kt]: Cleaned this up and used Kotlin Lambda

File: jetbrains-core/hs_err_pid28365.log [hunk 3]
```diff
@@ -0,0 +1,989 @@
+#
+# A fatal error has been detected by the Java Runtime Environment:
+#
+#  SIGSEGV (0xb) at pc=0x000000011541f05a, pid=28365, tid=33283
```
Comment (hunterwerlla) [inline jetbrains-core/hs_err_pid28365.log]: SIGSEGV

**Pull request merged**